### DISCUSSION TITLE
AceRoutine + Interrupts ?

### DISCUSSION THREAD

**Discussion by Goulaf** [bxparks/AceRoutine#33] [Q&A]
Hello,
Do any of you use interrupts along whith AceRoutine ?
I tried but it seems to crash some routines after a while.
Thanks !

**Comment by bxparks** [comment]
The `millis()` function is implemented using interrupts, so the short answer is, "Yes AceRoutine works with interrupts."

To figure out your problem, you need to upload a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example). Also, some info about the microcontroller that you are using.

**Comment by andrewspeirs** [comment]
Yes. I've used AceRoutine to control the speed of a stepper motor - the COROUTINE_DELAY controls the delay between pulses to the controller. I use this in combination with a hardware timer interrupt that controls data collection at regular intervals. The hardware timer is implemented in the SAMD51 Interrupt Timer library and calls the ISR at intervals specified in us.
